src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedArray = '';
+
+    if (this[0] !== null) {
+      joinedArray += this[0];
+    }
+
+    let tool = separator;
+
+    if (tool === null) {
+      tool = 'null';
+    }
```

Comment:
this way we will have nullls in the string, I don't think we want that

Reply:
I think we need nulls in the string for one of the tests:

test(`Should work with null`, () => {
  expect(source.join2(null))
    .toBe('0null1null2null3');
});

Reply:
It's true, but you should use template string to join the separator and the element, it will convert null to string by itself, you won't need to do it manually. Due to that you won't need to assign the separator variable to tool variable (prev comment)